### DISCUSSION TITLE
[FFmpeg-devel] [PATCH] libavcodec/qsvenc: Enable 444 encoding for RGB input

### DISCUSSION THREAD

**Discussion by alatteri** [CESNET/UltraGrid#283] [Ideas]
This should be something to keep an eye on, it will be a really nice addition to VPL as that becomes a more standard hardware acceleration on Intel processors.  Although it doesn't specify if this is 8bit or 10bit.

```
MSDK/VPL uses 420 chroma format as default to encode RGB, and this is
not a proper usage. Now enable 444 encoding for RGB input by default.
RGB is encoded using 444 chroma format when user doesn't specify the
profile or uses rext profile, otherwise, 420 is used.
```

https://ffmpeg.org/pipermail/ffmpeg-devel/2023-January/305242.html

**Comment by MartinPulec** [comment]
Thanks for the info. From the patch, I'd conclude that this is for 8-bit (`AV_PIX_FMT_BGRA` is 8-bit). Eg. QSV HEVC, however, may [support 10-bit as well](https://github.com/FFmpeg/FFmpeg/blob/fcd557a2c2174d89cb85630ef06c160027d52fa3/libavcodec/qsvenc_hevc.c#L401). But I really not sure if I understand it correctly, in the FFmpeg code, they already set chroma format for `q->param.mfx.FrameInfo.ChromaFormat`, but this could be just the input subsampling.
